src/main/java/ui/UI.java

```diff
@@ -427,6 +492,18 @@ public BrowserComponent getBrowserComponent() {
     }
 
     /**
+     * Returns the maximum usable size of the stage, or to the screen size if this fails.
+     */
+    public Rectangle getDimensions() {
+        Optional<Rectangle> dimensions = Utility.getUsableScreenDimensions();
+        if (dimensions.isPresent()) {
+            return dimensions.get();
+        } else {
+            return Utility.getScreenDimensions();
+        }
```

Comment:
Maybe we could put all of the screen-related methods in some kind of utility class? They're spread all over now.